compiler/backend/src/org/jetbrains/kotlin/codegen/AsmUtil.java

```diff
@@ -302,7 +302,8 @@ public static boolean isAbstractMethod(FunctionDescriptor functionDescriptor, Ow
     }
 
     public static boolean isStaticMethod(OwnerKind kind, CallableMemberDescriptor functionDescriptor) {
-        return isStaticKind(kind) ||
+        return isStaticDeclaration(functionDescriptor) ||
```

Comment:
Purpose of this change: Without this, the generated functions are not invoked with the correct descriptor; the enclosing class is included in the descriptor as the dispatch receiver parameter. This causes a stack underflow error during verification.

Reply:
I'm not sure if this change is safe to make since this affects the current non-IR backend, but all the tests are passing.

Reply:
It could be a bug in LocalDeclarationLowering, but i'm not sure.
In any way maybe it's best to postpone this pr for some time until new typeMapper will be appeared and KOTLIN-CR-2697 merged in master